### ISSUE TITLE
How did you run ropebwt2?

### ISSUE THREAD

**Issue by lh3** [KoslickiLab/prokrustean#17]
I was reading your manuscript (a good one btw), checked github and then noticed this sentence in your README:

> [ropebwt2] generates an incorrect data for fairly large datasets

I downloaded SRR20044276 and ran ropebwt2 and grlbwt on both read strands. The two tools gave identical BWT. In the early days, ropebwt2 and beetl could generate identical BWT on 100Gbp human reads. What command options were you using with ropebwt2? Note that ropebwt2 by default constructs BWT for both strands.

**Comment by akp6031** [comment]
I am sorry for the delayed response and thank you for your interest. I am so happy to hear your positive comment about the paper!
Following your earlier response in [issue #8](https://github.com/lh3/ropebwt2/issues/8) last year, I used the following command:
```
./ropebwt2 -R -o SRR20044276_rope.bwt SRR20044276.fastq.gz
```

I wonder if you have compared the outputs of grlbwt and ropebwt2 position by position. They match only up to position 2,132,355 but show significant differences beyond that. The following commands were used to compare these specific positions:
```
# This emits $
dd if=SRR20044276_grl.bwt bs=1 skip=$((2132355-1)) count=1 2>/dev/null
# This emits A
dd if=SRR20044276_rope.bwt bs=1 skip=$((2132355-1)) count=1 2>/dev/null
```

I could not install the BEETL yet, but I believe its equivalent computation as ropebwt2. My concern is, peculiarly, my function for recovering sequences from BWTs worked well for outputs of [optimalBWT](https://github.com/davidecenzato/optimalBWT), [grlbwt](https://github.com/ddiazdom/grlBWT), and [BCR_LCP_GSA](https://github.com/giovannarosone/BCR_LCP_GSA), but not with those from ropebwt2. Furthermore, when generating the Procrustean graph and counting k-mers, I obtained consistent results with the three libraries, except for ropebwt2.

I also conducted tests with other datasets. Results were consistent with small read sets around 10 MB but not with a larger set (40 MB) like [ERR10492989](https://www.ncbi.nlm.nih.gov/sra/?term=ERR10492989) with ropebwt2 against others. 

If you cannot easily identify a configuration issue on my end, I am eager to to assist with further testing! It would be great to adopt ropebwt2 as the default library because its performance is consistent on both macOS and Linux.

**Comment by lh3** [comment]
I guess this is caused by ambiguous bases in SRR20044276. Ropebwt2 follows A<C<G<T<N, even though the ascii of N is smaller than T. When preparing the input for grlBWT, I converted N to X instead. I then converted X back to N afterwards. grlBWT and ropebwt2 BWTs have the same checksum. Note that grlBWT and ropebwt3 (same core data structure as ropebwt2) produced identical BWT for 600Gb human pangenome data. Input size should not be a problem.
